server.go

```diff
@@ -153,6 +153,56 @@ func createPullSecret(clientset *kubernetes.Clientset, ns *corev1.Namespace, cre
 	return nil
 }
 
+// deletePullSecret deletes the image registry pull secret for the provided namespace
+func deletePullSecret(clientset *kubernetes.Clientset, ns corev1.Namespace) error {
+	secrets := clientset.CoreV1().Secrets(ns.Name)
+	if err := secrets.Delete(context.TODO(), gcpAuth, metav1.DeleteOptions{}); err != nil {
+		return fmt.Errorf("deleting %s secret in %s namespace: %v", gcpAuth, ns.Name, err)
+	}
+	return nil
+}
+
+// refreshAllPullSecrets deletes and recreates image registry pull secrets for all namespaces
+func refreshAllPullSecrets() error {
+	creds, err := google.FindDefaultCredentials(context.Background())
+	if err != nil {
+		return fmt.Errorf("finding default credentials: %v", err)
+	}
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		return fmt.Errorf("getting cluster config: %v", err)
+	}
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return fmt.Errorf("getting clientset: %v", err)
+	}
+	namespaceList, err := clientset.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("listing namespaces: %v", err)
+	}
+	for _, ns := range namespaceList.Items {
+		if ns.Name == "kube-system" || ns.Name == "gcp-auth" {
+			continue
+		}
+		if err := deletePullSecret(clientset, ns); err != nil {
+			log.Println(err)
+		}
+		if err := createPullSecret(clientset, &ns, creds); err != nil {
+			log.Println(err)
+		}
+	}
+	return nil
+}
+
+// pullSecretTicker refreshes all the image registry pull secrets every hour
+func pullSecretTicker() {
+	for range time.Tick(1 * time.Hour) {
```

Comment:
this might be too aggressive of a refresh, for someone who might not need it

Reply:
Changed to every six hours

Reply:
Isn't the default expiration set to 1 hour? In our case, a six-hour interval would _not_ work. I feel it would be best to stay aligned with this default value. Can we make this configurable with a default 1 hour time period?

Reply:
I looked it up and tokens do indeed have a default of one hour, I'll revert the change